pkg/agent/agent.go

```diff
@@ -507,13 +440,26 @@ func (a *Agent) watch(ctx context.Context, snapshotURL, diagnosticsURL string, c
 		case directive := <-a.newDirective:
 			a.directiveHandler.HandleDirective(ctx, a, directive)
 		}
+
 		// only ask ambassador for a snapshot if we're actually going to report it.
 		// if reportRunning is true, that means we're still in the quiet period
 		// after sending a report.
+		// if emissary is the owner, do all the things
 		if !a.reportingStopped && !a.reportRunning.Value() {
-			snapshot, err := getAmbSnapshotInfo(snapshotURL)
-			if err != nil {
-				dlog.Warnf(ctx, "Error getting snapshot from ambassador %+v", err)
+			// if emissary is present, get initial snapshot from emissary
+			// otherwise, create it
+			var snapshot *snapshotTypes.Snapshot
+			if a.emissaryPresent {
```

Comment:
It looks like this is never actually set anywhere.

Reply:
good catch, ty